tests/examples/modes/on-change.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { test, testWithoutJS, expect } from 'tests/setup/tests'
+
+const route = '/examples/modes/on-change'
+
+test('With JS enabled', async ({ example }) => {
+  const { firstName, email, button, page } = example
+
+  await page.goto(route)
+
+  // Render
+  await example.expectField(firstName)
+  await example.expectField(email)
+  await expect(button).toBeDisabled()
+
+  // Client-side validation
+  await firstName.input.type('a')
+  await firstName.input.fill('')
+
+  await example.expectError(
+    firstName,
+    'String must contain at least 1 character(s)',
+  )
+
+  await email.input.type('a')
+  await email.input.fill('')
+  await example.expectError(
+    email,
+    'String must contain at least 1 character(s)',
+  )
+  await email.input.fill('john')
+  await example.expectError(email, 'Invalid email')
+
+  // Make form be valid
+  await firstName.input.fill('John')
+  await email.input.fill('john@doe.com')
+  await expect(button).toBeEnabled()
+
+  // Submit form
+  await button.click()
+
+  const actionResult = await page
+    .locator('#action-data > pre:visible')
+    .innerText()
+
+  await expect(actionResult).toContain('"firstName": "John"')
+  await expect(actionResult).toContain('"email": "john@doe.com"')
```

Comment:
Should we test anything else? It didn't seem to make sense to test autoFocus here (if possible?)